portfolio/src/main/webapp/script.js

```diff
@@ -21,20 +21,31 @@ function getNameUsingArrowFunctions() {
     });
 }
 
+
 /**
  * Fetches stats from the servers and adds them to the DOM.
  */
 function getDataComment() {
-  fetch('/data').then(response => response.json()).then((dataComments) => {
+    fetch('/data').then(response => response.json()).then((dataComments) => {
       const commentContainer = document.getElementById("dataComment-container");
-      dataComments.forEach(dataComment => { 
+      const numOfCommentsToDisplay = document.getElementById("numComments").value;
+      
+      dataComments.splice(0, numOfCommentsToDisplay).forEach(dataComment => { 
           let miniP = document.createElement("p");
           miniP.innerText = "Basketball Player: " + dataComment.message;
           commentContainer.appendChild(miniP);
       })
   });
 }
 
+/**
+ * Takes in user input's choice of number of comments to display.
+ */
+function getDiffNumberOfDataComments() {
```

Comment:
looks like this function only resets the innerText and calls getDataComment(), which is unnecessary. There is a better way to organize the code. But for now, I think you can move the “innerText" line to getDataComment, and remove this function. Put getDataComment as the callback for input's onchange

Reply:
done, I've been trying to explore different ways to organize the code too - I made the change for now but will definitely come back to find the most optimal way.